engine/src/execution_graph/logic_controllers/LogicalFilter.cpp

```diff
@@ -410,16 +410,23 @@ std::unique_ptr<ral::frame::BlazingTable> processJoin(
   }
 
   if(join_type == INNER_JOIN) {
+    //Removing nulls on key columns before joining
+    auto table_left_dropna = cudf::experimental::drop_nulls(table_left.view(), left_column_indices);
```

Comment:
We need to check if any of the key columns have any nulls before calling `cudf::experimental::drop_nulls` and only call that funcion if necessary

Reply:
I updated the code, please, check it again.